src/test/Makefile

```diff
@@ -438,11 +441,28 @@ COMMON_FLAGS = \
 	-O0 \
 	-DUNIT_TEST \
 	-isystem $(GTEST_DIR)/inc \
-	-MMD -MP
+	-MMD -MP \
+	-Wno-c99-extensions \
+	-Wno-reorder
 
 CC_VERSION = $(shell $(CC) -dumpversion)
 CXX_VERSION = $(shell $(CXX) -dumpversion)
 
+# Please revisit versions when new clang version arrive.
+CC_VERSION_MAJOR := $(shell expr `$(CC) -dumpversion | cut -f1 -d.`)
+CC_VERSION_CHECK_MIN := 7
+# Add 1 to version requested
+CC_VERSION_CHECK_MAX := 11
+ifdef MACOSX
+CC_VERSION_CHECK_MAX := 13
```

Comment:
What versions of MacOS report clang versions 11 or 12? Also, have we tested with these?
Again, the naming is off here, `CC_VERSION_CHECK_MAX` is **not** the maximum allowable version, so it should not be named this way.
I'd skip the 1 offset and then use a 'greater than' comparison below: `$(CC_VERSION_MAJOR) \> $(CC_VERSION_CHECK_MAX)`

Reply:
I found these from the clang -dumpversion from output of Catalina in the other PR?

Reply:
Your line doesn't need the offset 👍